### DISCUSSION TITLE
How does zellij decide position of panel?

### DISCUSSION THREAD

**Discussion by al3xandru** [zellij-org/zellij#1348] [Q&A]
I'm running the latest zellij on macOS. I have a Terminal.app window and an iTerm window. zellij running on both.

1. In Terminal.app, first `Alt+n` creates the panel underneath (horizontally). 
2. In iTerm, first `Alt+n` creates the panel vertically. 

There are no different configurations. It doesn't seem to be related to the size of the windows either. What leads to this?

thank you

**Comment by imsnif** [comment]
When Zellij starts, it queries the terminal emulator for the size in pixels of the character cell. It then uses this size to find the largest free space on screen.

Some terminal emulators don't support this query, and in that case Zellij guesstimates the size of a character cell (or rather, the ratio of its width to height, which is what it really needs) and uses that. I guess maybe that can explain the difference? Otherwise it could be things like font and window size which affect the count of cells in said free space.

**Reply by al3xandru** [reply]
@imsnif thanks for the explanations.

1. Based on your experience do you know if iTerm2 or Terminal app report this information similarly?
2. Whatever resizing I do, I don't seem to be able to reproduce the same series of 
3. I have upgraded to 0.28.1 from 0.26.1 a few days ago. I have noticed this behavior right after this update.

**Comment by al3xandru** [comment]
I don't know how to reproduce this behavior but here is what a series of `Alt+N` creates now:

<img width="809" alt="image" src="https://user-images.githubusercontent.com/290841/165371681-45a01f1e-0373-4af0-a15a-800a90259244.png">
